mmdet/datasets/pipelines/transforms.py

```diff
@@ -2756,3 +2756,149 @@ def __repr__(self):
+        self.mask_occluded_thresh = mask_occluded_thresh
+
+    def get_indexes(self, dataset):
+        """Call function to collect indexes.s.
+
+        Args:
+            dataset (:obj:`MultiImageMixDataset`): The dataset.
+        Returns:
+            list: indexes.
+        """
+        return random.randint(0, len(dataset))
+
+    def __call__(self, results):
+        """Call function to make a copy-paste of image.
+
+        Args:
+            results (dict): Result dict.
+        Returns:
+            dict: Result dict with copy-paste transformed.
+        """
+
+        assert 'mix_results' in results
+        assert len(results['mix_results']
+                   ) == 1, 'Copy-paste only support 2 images now !'
+
+        selected_results = self._select_object(results['mix_results'][0])
+        return self._copy_paste(results, selected_results)
+
+    def _select_object(self, results):
+        bboxes = results['gt_bboxes']
+        labels = results['gt_labels']
+        masks = results['gt_masks']
+        max_paste_objects = min(bboxes.shape[0] + 1, self.max_paste_objects)
+        num_paste_objects = np.random.randint(0, max_paste_objects)
+        selected_inds = np.random.choice(
+            bboxes.shape[0], size=num_paste_objects, replace=False)
+
+        selected_bboxes = bboxes[selected_inds]
+        selected_labels = labels[selected_inds]
+        selected_masks = self.get_bitmapmasks(masks.masks[selected_inds])
+
+        results['gt_bboxes'] = selected_bboxes
+        results['gt_labels'] = selected_labels
+        results['gt_masks'] = selected_masks
+        return results
+
+    def _copy_paste(self, dst_results, src_results):
+        dst_img = dst_results['img']
+        dst_bboxes = dst_results['gt_bboxes']
+        dst_labels = dst_results['gt_labels']
+        dst_masks = dst_results['gt_masks']
+
+        src_img = src_results['img']
+        src_bboxes = src_results['gt_bboxes']
+        src_labels = src_results['gt_labels']
+        src_masks = src_results['gt_masks']
+
+        if len(src_bboxes) == 0:
+            return dst_results
+
+        # update masks and generate bboxes from updated masks
+        compose_mask = np.where(np.any(src_masks.masks, axis=0), 1, 0)
+        updated_dst_masks = self.get_updated_masks(dst_masks, compose_mask)
+        updated_dst_bboxes = updated_dst_masks.get_bboxes()
+        assert len(updated_dst_bboxes) == len(updated_dst_masks)
+
+        # filter some objects which are totally occluded
+        included_bboxes_inds = np.all(
+            np.abs((updated_dst_bboxes - dst_bboxes)) <=
+            self.bbox_occluded_thresh,
+            axis=-1)
+        included_masks_inds = updated_dst_masks.masks.sum(
+            axis=(1, 2)) > self.mask_occluded_thresh
+        included_inds = included_bboxes_inds | included_masks_inds
+
+        # Paste source objects to destination image directly
+        included_img = dst_img * (1 - compose_mask[..., np.newaxis]
```

Comment:
Shall we add a method interface for image blending before pasting the image so that the users could implement different blending functions.